hash.c

```diff
@@ -1810,22 +1810,69 @@ rb_hash_initialize(int argc, VALUE *argv, VALUE hash)
 
 /*
  *  call-seq:
- *     Hash[ key, value, ... ]         -> new_hash
- *     Hash[ [ [key, value], ... ] ]   -> new_hash
- *     Hash[ object ]                  -> new_hash
+ *    Hash[] -> new_empty_hash
+ *    Hash[*keys_and_values] -> new_hash
```

Comment:
I think the original signature is a bit clearer?

Reply:
Original uses square brackets in two ways in one signature:  repeatable element (with ellpisis) and actual Array literal.  If I don't hear, I'll change to original.

Reply:
`* keys_and_values` doesn't convey the same level of detail IMHO.

However rather than reverting to the original, can we make some tweak to fix the issue you identified and also keep it clear the structure of the arguments?